docs/reference/components/sinks/redis.cue

```diff
@@ -0,0 +1,127 @@
+package metadata
+
+components: sinks: redis: {
+	title: "Redis"
+	classes: {
+		commonly_used: false
+		delivery:      "best_effort"
+		development:   "beta"
+		egress_method: "stream"
+		service_providers: []
+		stateful: false
+	}
+	features: {
+		buffer: enabled:      false
+		healthcheck: enabled: true
+		send: {
+			compression: enabled: false
+			encoding: {
+				enabled: true
+				codec: {
+					enabled: true
+					default: "json"
+					enum: ["json", "text"]
+				}
+			}
+			request: enabled: false
+			tls: {
+				enabled:                true
+				can_enable:             true
+				can_verify_certificate: false
+				can_verify_hostname:    false
+				enabled_default:        false
+			}
+			to: {
+				service: services.redis
+				interface: {
+					socket: {
+						direction: "outgoing"
+						protocols: ["tcp"]
+						ssl: "optional"
+					}
+				}
+			}
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	configuration: {
+		url: {
+			description: "The Redis URL to connect to. The url _must_ take the form of `protocol://server:port/db` where the protocol can either be `redis` or `rediss` for connections secured via TLS."
+			groups: ["tcp"]
+			required: true
+			warnings: []
+			type: string: {
+				examples: ["redis://127.0.0.1:6379/0"]
+				syntax: "literal"
+			}
+		}
+		key: {
+			description: "The Redis key to publish messages to."
+			required:    true
+			warnings: []
+			type: string: {
+				examples: ["syslog:{{ app }}", "vector"]
+				syntax: "template"
+			}
+		}
+		data_type: {
+			common:      false
+			description: "The Redis data type (`list` or `channel`) to use."
+			required:    false
+			type: string: {
+				default: "list"
+				enum: {
+					list:    "Use the Redis `list` data type."
+					channel: "Use the Redis `channel` data type."
+				}
+				syntax: "literal"
+			}
+		}
+		method: {
```

Comment:
I've been pushing for us to namespace these "mode" specific options. I'm imagining that we might want additional methods in the future that are specific to new data types.

What would you think of allowing the configuration to look like:

```toml
[sinks.out]
type = "redis"
url = "redis://127.0.0.1:6379/0"
key = "vector"
data_type = "list"
list.method = "lpush"
healthcheck = true
inputs = ["in"]
encoding.codec = "json"
```

?